src/OpenTelemetry.Instrumentation.AWSLambda/Implementation/HttpSemanticConventions.cs

```diff
@@ -0,0 +1,140 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using Amazon.Lambda.APIGatewayEvents;
+using OpenTelemetry.Trace;
+
+namespace OpenTelemetry.Instrumentation.AWSLambda.Implementation
+{
+    internal class HttpSemanticConventions
+    {
+        // x-forwarded-... headres are described here https://docs.aws.amazon.com/elasticloadbalancing/latest/classic/x-forwarded-headers.html
+        private const string HeaderXForwardedProto = "X-Forwarded-Proto";
+        private const string HeaderHost = "Host";
+
+        internal static IEnumerable<KeyValuePair<string, object>> GetHttpTags<TInput>(TInput input)
+        {
+            var tags = new List<KeyValuePair<string, object>>();
+
+            string httpScheme = null;
+            string httpTarget = null;
+            string httpMethod = null;
+            string hostName = null;
+            int? hostPort = null;
+
+            switch (input)
+            {
+                case APIGatewayProxyRequest request:
+                    httpScheme = GetHeaderValue(request, HeaderXForwardedProto);
+                    httpTarget = request.Path;
+                    httpMethod = request.HttpMethod;
+                    var hostHeader = GetHeaderValue(request, HeaderHost);
+                    (hostName, hostPort) = GetHostAndPort(httpScheme, hostHeader);
+                    break;
+                case APIGatewayHttpApiV2ProxyRequest requestV2:
+                    httpScheme = GetHeaderValue(requestV2, HeaderXForwardedProto);
+                    httpTarget = requestV2?.RequestContext?.Http?.Path;
+                    httpMethod = requestV2?.RequestContext?.Http?.Method;
+                    var hostHeaderV2 = GetHeaderValue(requestV2, HeaderHost);
+                    (hostName, hostPort) = GetHostAndPort(httpScheme, hostHeaderV2);
+                    break;
+            }
+
+            tags.AddTagIfNotNull(SemanticConventions.AttributeHttpScheme, httpScheme);
+            tags.AddTagIfNotNull(SemanticConventions.AttributeHttpTarget, httpTarget);
+            tags.AddTagIfNotNull(SemanticConventions.AttributeHttpMethod, httpMethod);
+            tags.AddTagIfNotNull(SemanticConventions.AttributeNetHostName, hostName);
+            tags.AddTagIfNotNull(SemanticConventions.AttributeNetHostPort, hostPort);
+
+            return tags;
+        }
+
+        internal static void SetHttpTagsFromResult<TResult>(Activity activity, TResult result)
```

Comment:
```suggestion
        internal static void SetHttpTagsFromResult(Activity activity, object result)
```

It does not look like the generic buys us anything here.

Reply:
Changed to object and committed along with other changes.